lib/src/front_end/piece_factory.dart

```diff
@@ -408,29 +408,46 @@ mixin PieceFactory implements CommentWriter {
     pieces.give(builder.build());
   }
 
-  /// Creates a class, enum, extension, etc. declaration with a body containing
-  /// members.
-  void createType(
-      NodeList<Annotation> metadata,
-      List<Token?> modifiers,
-      Token keyword,
-      Token name,
-      TypeParameterList? typeParameters,
+  /// Creates a class, enum, extension, mixin, or mixin application class
+  /// declaration.
+  ///
+  /// For all but a mixin application class, [body] should a record containing
+  /// the bracket delimiters and the list of member declarations for the type's
+  /// body.
+  ///
+  /// For mixin application classes, [body] is `null` and instead [equals],
+  /// [superclass], and [semicolon] are provided.
```

Comment:
can these be split into different calls with more statically useful signatures (that might forward to a shared private implementation)

Reply:
I thought about that, but most of the parameters are valid in both so there would be a lot of pointless forwarding. And the function for handling mixin application classes would only be called once. This seemed the least boilerplate-y to me.